### PR TITLE
chore: Standardize TutorialsPoint resources

### PR DESCRIPTION
**Pull request by davorpa**
## What does this PR do?
Add info | Improve repo

### Description

Standardize TutorialsPoint  links & metadatas: 
- Urls: use top most path that works, mainly remove `index.htm` from them
- Author: `Tutorials Point` is the right name
- Resource format: HTML & PDF

### Why is this valuable (or not)?

Resolves pending task about this kind of resources added to hacktoberfest backlog #5762 

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -381,7 +381,7 @@
 ### Docker
 
 * [Docker](https://www.youtube.com/playlist?list=PLhW3qG5bs-L99pQsZ74f-LC-tOEsBp2rK) - Raghav Pal
-* [Docker Tutorial](https://www.tutorialspoint.com/docker/index.htm) - TutorialsPoint
+* [Docker Tutorial](https://www.tutorialspoint.com/docker/) - Tutorials Point (HTML, PDF)
```
Comment (LuigiImVector) [inline courses/free-courses-en.md:384]:
```suggestion
* [Docker Tutorial](https://www.tutorialspoint.com/docker/) - Tutorials Point (HTML, [PDF](https://www.tutorialspoint.com/docker/docker_tutorial.pdf))
```

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -439,7 +439,7 @@
 
 * [A Tour Of Go](https://tour.golang.org/welcome/1)
 * [DevOps BootCamp](https://github.com/jeffotoni/goworkshopdevops) - Jefferson Otoni Lima et al.
-* [Go Tutorial](http://www.tutorialspoint.com/go/index.htm)
+* [Go Tutorial](http://www.tutorialspoint.com/go/) - Tutorials Point (HTML, PDF)
```
Comment (LuigiImVector) [inline courses/free-courses-en.md:442]:
```suggestion
* [Go Tutorial](http://www.tutorialspoint.com/go/) - Tutorials Point (HTML, [PDF](https://www.tutorialspoint.com/go/go_tutorial.pdf))
```

File: courses/free-courses-en.md [hunk 3]
```diff
@@ -697,7 +697,7 @@
 
 * [Learn Lua in 15 Minutes](http://tylerneylon.com/a/learn-lua/) - Tyler Neylon (HTML)
 * [Lua Interactive Crash Course](https://web.archive.org/web/20201111225216/luatut.com/crash_course.html)
-* [Lua Tutorial](http://www.tutorialspoint.com/lua/)
+* [Lua Tutorial](http://www.tutorialspoint.com/lua/) - Tutorials Point (HTML, PDF)
```
Comment (LuigiImVector) [inline courses/free-courses-en.md:700]:
```suggestion
* [Lua Tutorial](http://www.tutorialspoint.com/lua/) - Tutorials Point (HTML, [PDF](https://www.tutorialspoint.com/lua/lua_tutorial.pdf))
```

File: courses/free-courses-en.md [hunk 4]
```diff
@@ -880,7 +880,7 @@
 #### Flask
 
 * [Flask Tutorials](https://www.youtube.com/playlist?list=PL-osiE80TeTs4UjLw5MM6OjgkjFeUxCYH) - Corey Schafer
-* [Python Flask Tutorial](https://www.tutorialspoint.com/flask/flask_overview.htm) (TutorialsPoint)
+* [Python Flask Tutorial](https://www.tutorialspoint.com/flask/) - Tutorials Point (HTML, PDF)
```
Comment (LuigiImVector) [inline courses/free-courses-en.md:883]:
```suggestion
* [Python Flask Tutorial](https://www.tutorialspoint.com/flask/) - Tutorials Point (HTML, [PDF](https://www.tutorialspoint.com/flask/flask_tutorial.pdf))
```

File: books/free-programming-books-subjects.md [hunk 5]
```diff
@@ -351,7 +351,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Machine Learning for Data Streams](https://moa.cms.waikato.ac.nz/book-html/) - Albert Bifet, Ricard Gavaldà, Geoff Holmes, Bernhard Pfahringer
 * [Machine Learning from Scratch](https://dafriedman97.github.io/mlbook/) - Danny Friedman (HTML, PDF, Jupyter Book)
 * [Machine Learning, Neural and Statistical Classification](http://www1.maths.leeds.ac.uk/~charles/statlog/)
-* [Machine Learning with Python](https://www.tutorialspoint.com/machine_learning_with_python) - Tutorials Point (HTML, [PDF](https://www.tutorialspoint.com/machine_learning_with_python/machine_learning_with_python_tutorial.pdf))
+* [Machine Learning with Python](https://www.tutorialspoint.com/machine_learning_with_python/) - Tutorials Point (HTML, [PDF](https://www.tutorialspoint.com/machine_learning_with_python/machine_learning_with_python_tutorial.pdf))
```
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-subjects.md:354]: > We can standardize Tutorials Point resources like this

Hi Victor. I agree with you about using multiformat links. What do you think @eshellman ?

[the remark this answers is not in the document]
Comment (eshellman) [reply books/free-programming-books-subjects.md:354]: one link per resource please. Unless the second link is not related to the first. Multiple links are more than twice as hard to maintain.

[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-subjects.md:354]: > one link per resource please. Unless the second link is not related to the first. Multiple links are more than twice as hard to maintain.

Then resolved without applying

File: courses/free-courses-en.md [hunk 6]
```diff
@@ -115,7 +115,7 @@
 * [Data Structures and Algorithms](https://www.youtube.com/playlist?list=PLdo5W4Nhv31bbKJzrsKfMpo_grxuLl8LU) - Jenny's lectures CS/IT NET&JRF
 * [Data Structures and Algorithms for Beginners](https://www.youtube.com/watch?v=BBpAmxU_NQo) - Moshfegh Hamedani (Programming with Mosh)
 * [Data Structures and Algorithms Specialization](https://www.coursera.org/specializations/data-structures-algorithms) - UC San Diego, HSE University
-* [Data Structures and Algorithms Tutorial](https://www.tutorialspoint.com/data_structures_algorithms/) - Tutorials Point
+* [Data Structures and Algorithms Tutorial](https://www.tutorialspoint.com/data_structures_algorithms/) - Tutorials Point (HTML, PDF)
```
[the remark this answers is not in the document]
Comment (davorpa) [reply courses/free-courses-en.md:118]: There are 5 of this kind. I think needs be moved to subjects and langs too. Are you agree @eshellman ?

File: books/free-programming-books-langs.md [hunk 7]
```diff
@@ -258,7 +258,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 
 * [Android Notes for Professionals](https://goalkicker.com/AndroidBook) - Compiled from StackOverflow Documentation (PDF)
 * [Android Programming Succinctly, Syncfusion](https://www.syncfusion.com/resources/techportal/ebooks/android) ([PDF](https://www.syncfusion.com/Account/Logon?ReturnUrl=%2fresources%2ftechportal%2febooks%2fandroid), [Kindle](https://www.syncfusion.com/Account/Logon?ReturnUrl=%2fresources%2ftechportal%2febooks%2fandroid)) (email address *requested*, not required)
-* [Android Tutorial](http://www.tutorialspoint.com/android) - Tutorials Point
+* [Android Tutorial](http://www.tutorialspoint.com/android/) - Tutorials Point (HTML, PDF)
```
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-langs.md:261]: See https://github.com/EbookFoundation/free-programming-books/pull/6799#discussion_r836791385 for resolution

File: courses/free-courses-en.md (hunk 1 above)
Comment (davorpa) [reply courses/free-courses-en.md:384]: See https://github.com/EbookFoundation/free-programming-books/pull/6799#discussion_r836791385 for resolution

File: courses/free-courses-en.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (eshellman) [reply courses/free-courses-en.md:118]: IIRC, TutorialsPoint had 2 different sorts of resources, one of which was more course-like than the other

[the remark this answers is not in the document]
Comment (eshellman) [reply courses/free-courses-en.md:118]: If thats not true anymore, it's worth reconsidering

[the remark this answers is not in the document]
Comment (davorpa) [reply courses/free-courses-en.md:118]: Then transfering those 5 of 7 resources

**Review by LuigiImVector — APPROVED** [review]

**Review by CleanMachine1 — APPROVED** [review]

**Pull request merged**